From 4283331711b135601a2ff3dcaa358c5ae5638657 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 1 Sep 2014 20:15:28 -0400
Subject: [Documentation] powerpc/perf/hv_{gpci, 24x7}: Add documentation of device attributes

Message-id: <1409602534-23951-10-git-send-email-gduarte@redhat.com>
Patchwork-id: 90123
O-Subject: [RHEL7.1 PATCH BZ 947159 09/15] powerpc/perf/hv_{gpci, 24x7}: Add documentation of device attributes
Bugzilla: 947159
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=947159
Upstream Status: ad0a02b834777f2e107bfd2f3fa240cd7335cc59

commit ad0a02b834777f2e107bfd2f3fa240cd7335cc59
Author: Cody P Schafer <cody@linux.vnet.ibm.com>
Date:   Fri Mar 14 16:00:44 2014 +1100

    powerpc/perf/hv_{gpci, 24x7}: Add documentation of device attributes

    gpci and 24x7 expose some device specific attributes. Add some
    documentation for them.

    Signed-off-by: Cody P Schafer <cody@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_24x7 b/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_24x7
new file mode 100644
index 0000000..e78ee79
--- /dev/null
+++ b/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_24x7
@@ -0,0 +1,23 @@
+What:  /sys/bus/event_source/devices/hv_24x7/interface/catalog
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  Provides access to the binary "24x7 catalog" provided by the
+  hypervisor on POWER7 and 8 systems. This catalog lists events
+  avaliable from the powerpc "hv_24x7" pmu. Its format is
+  documented here:
+  https://raw.githubusercontent.com/jmesmon/catalog-24x7/master/hv-24x7-catalog.h
+
+What:  /sys/bus/event_source/devices/hv_24x7/interface/catalog_length
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  A number equal to the length in bytes of the catalog. This is
+  also extractable from the provided binary "catalog" sysfs entry.
+
+What:  /sys/bus/event_source/devices/hv_24x7/interface/catalog_version
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  Exposes the "version" field of the 24x7 catalog. This is also
+  extractable from the provided binary "catalog" sysfs entry.
diff --git a/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_gpci b/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_gpci
new file mode 100644
index 0000000..3fa58c2
--- /dev/null
+++ b/Documentation/ABI/testing/sysfs-bus-event_source-devices-hv_gpci
@@ -0,0 +1,43 @@
+What:  /sys/bus/event_source/devices/hv_gpci/interface/collect_privileged
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  '0' if the hypervisor is configured to forbid access to event
+  counters being accumulated by other guests and to physical
+  domain event counters.
+  '1' if that access is allowed.
+
+What:  /sys/bus/event_source/devices/hv_gpci/interface/ga
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  0 or 1. Indicates whether we have access to "GA" events (listed
+  in arch/powerpc/perf/hv-gpci.h).
+
+What:  /sys/bus/event_source/devices/hv_gpci/interface/expanded
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  0 or 1. Indicates whether we have access to "EXPANDED" events (listed
+  in arch/powerpc/perf/hv-gpci.h).
+
+What:  /sys/bus/event_source/devices/hv_gpci/interface/lab
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  0 or 1. Indicates whether we have access to "LAB" events (listed
+  in arch/powerpc/perf/hv-gpci.h).
+
+What:  /sys/bus/event_source/devices/hv_gpci/interface/version
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  A number indicating the version of the gpci interface that the
+  hypervisor reports supporting.
+
+What:  /sys/bus/event_source/devices/hv_gpci/interface/kernel_version
+Date:  February 2014
+Contact: Cody P Schafer <cody@linux.vnet.ibm.com>
+Description:
+  A number indicating the latest version of the gpci interface
+  that the kernel is aware of.
-- 
1.7.1